client/src/Components/Layouts/Customers/index.jsx

```diff
@@ -92,35 +92,90 @@ export default class Customers extends Component {
             filteredcustomersName: filterCustomerName
         })
     }
+
+
+    showModal = () => {
+        this.setState({ visible: true });
+    };
+    handleCancel = () => {
+        this.setState({ visible: false });
+    };
+
+    handleCreate = () => {
+        const form = this.formRef.props.form;
+        form.validateFields((err, values) => {
+            if (err) {
+                return;
+            }
+            let addCustomer = {
+                name: values.name,
+                email: values.email,
+                phone: parseInt(values.prefixPhone + values.phone),
+                status: values.status,
+                address: values.address
+            }
+            fetch('api/v1/addcustomer', {
+                method: 'POST',
+                headers: { 'content-type': 'application/json' },
+                body: JSON.stringify(addCustomer)
+            }).then(res => res.json())
+                .then(result => {
+                    if (result.result) {
+                        let newcustomer = [...this.state.customers];
+                        newcustomer.push(result.result[0])
+                        let newallData = [...this.state.customers];
+                        newallData.push(result.result[0])
+                        this.setState({
+                            customers: newcustomer,
+                            allData: newallData
+                        })
+                    }
+                })
+            form.resetFields();
+            this.setState({ visible: false });
```

Comment:
this line is repeated more than the one you can put it in one function and use it : 
```
visibilityHandle = () => {
this.setState(prev => {
return {visible: !prev.visible};
})
}
```
and call it in `handleCancel, showModal` and here